doc/nrf/gs_testing.rst

```diff
@@ -120,3 +120,29 @@ To connect to the nRF9160-based kit with LTE Link Monitor, perform the following
    .. note::
 
       In the case of nRF9160 DK, the reset button must be pressed to restart the device and to start the application.
+
+.. _gs_debugging:
+
+Debugging an application
+************************
+
+Given that |VSC| is the main IDE for the |NCS|, set up the debug session as described in `Debugging an application`_ in the |nRFVSC| documentation.
+
+To debug the application core firmware, you need a single debug session.
+To debug the firmware running on the network core, you also need to set up a separate debug session for the application core.
+
+Complete the following steps to start debugging the network core:
+
+1. Set up sessions for the application core and network core as mentioned in `Debugging an application`_.
+#. Select the appropriate CPU for debugging in each session, the nRF5340 application core and the nRF5340 network core respectively.
+#. Once both sessions are established, execute the code on the application core.
+
+   The startup code releases the ``NETWORK.FORCEOFF`` signal to start the network core and allocates the necessary GPIO pins for it.
+#. Start code execution on the network core in the other debug session.
+
+If you want to reset the network core while debugging, make sure to first reset the application core and execute the code.
+
+You can also use the GDB command line tool for debugging.
+
+.. note::
+  Debugging firmware on the application core in the non-secure domain (when using the ``nrf5340dk_nrf5340_cpuapp_ns`` build target), is currently not supported.
```

Comment:
Probably want to generalize this as well, to something like "(when using a build target that ends in \`\`_ns\`\`)".